doc/shared/sunstepper/SUNStepper_Implementing.rst

```diff
@@ -0,0 +1,56 @@
+.. ----------------------------------------------------------------
+   Programmer(s): David J. Gardner @ LLNL
+                  Steven B. Roberts @ LLNL
+   ----------------------------------------------------------------
+   SUNDIALS Copyright Start
+   Copyright (c) 2002-2024, Lawrence Livermore National Security
+   and Southern Methodist University.
+   All rights reserved.
+
+   See the top-level LICENSE and NOTICE files for details.
+
+   SPDX-License-Identifier: BSD-3-Clause
+   SUNDIALS Copyright End
+   ----------------------------------------------------------------
+
+.. _SUNStepper.Implementing:
+
+Implementing a SUNStepper
+=========================
```

Comment:
It would be great if we had an example that implemented a custom SUNStepper, in which case this file could also mention an example where users can see these instructions "in action."  I'm imagining a simple operator-splitting example where one partition has an analytical solution.

Reply:
In the operator splitting PR, currently only a unit test creates a custom SUNStepper. I can see about modifying the advection-diffusion-reaction example so we have some something to point to.

Reply:
Or this could be added in the near future (not required for approval of this PR).